framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/api/Disambiguation.java

```diff
@@ -0,0 +1,160 @@
+     */
+    @JsonCreator
+    public Disambiguation(@JsonProperty("abbreviation") @NotNull String abbreviation, @JsonProperty("meanings") @NotNull String[] meanings) {
+        this(abbreviation, new LinkedHashSet<>(Arrays.stream(meanings).toList()));
+    }
+
+    /**
+     * Creates a new disambiguation of the provided abbreviation using the given set of meanings.
+     *
+     * @param abbreviation the abbreviation that is disambiguated by this instance
+     * @param meanings     a set of meanings for the abbreviation, may be empty
+     */
+    public Disambiguation(@NotNull String abbreviation, @NotNull LinkedHashSet<String> meanings) {
+        this.abbreviation = abbreviation;
+        this.meanings = meanings;
+    }
+
+    /**
+     * Adds all meanings from another disambiguation to this disambiguation. Be careful, this does not perform any checks regarding the abbreviations.
+     *
+     * @param other the other disambiguation
+     * @return this
+     */
+    public @NotNull Disambiguation addMeanings(Disambiguation other) {
+        meanings.addAll(other.meanings);
+        return this;
+    }
+
+    /**
+     * Searches the text for meanings contained by this disambiguation and replaces them with the abbreviation.
+     *
+     * @param text       the text to search
+     * @param ignoreCase whether letter case should be ignored when searching for the meanings
+     * @return the abbreviated text
+     */
+    public @NotNull String replaceMeaningWithAbbreviation(@NotNull String text, boolean ignoreCase) {
+        var abbreviatedText = text;
+        for (String meaning : meanings) {
+            String pattern = ignoreCase ? "(?i)" : "";
+            pattern += meaning;
+            abbreviatedText = abbreviatedText.replaceAll(pattern, abbreviation);
+        }
+        return abbreviatedText;
+    }
+
+    /**
+     * Creates a map with entries mapping each abbreviation to its disambiguation. This is useful if a list of disambiguations has to be searched for a
+     * particular abbreviation regularly.
+     *
+     * @param disambiguations a list of disambiguations, may be empty
+     * @return an immutable map
+     */
+    public @NotNull static ImmutableMap<String, Disambiguation> toMap(@NotNull List<Disambiguation> disambiguations) {
+        var map = Maps.mutable.<String, Disambiguation>empty();
+        for (var disambiguation : disambiguations) {
+            map.merge(disambiguation.getAbbreviation(), disambiguation, Disambiguation::addMeanings);
+        }
+        return map.toImmutable();
+    }
+
+    /**
+     * Merges the first map with the second map in a new map. If a key already exists, the disambiguations are merged non-destructively.
+     *
+     * @param a the first map
+     * @param b the second map
+     * @return a mutable map
+     */
+    @NotNull
+    public static MutableMap<String, Disambiguation> merge(@NotNull Map<String, Disambiguation> a, @NotNull Map<String, Disambiguation> b) {
```

Comment:
The signature has been adapted to use (Unmodifiable-)LinkedHashMaps. I can change it if you want sorted parameters and return values, but if the concern was mainly about determinism, this may be sufficient